File: src/binary/gates/gates.go
package gates

func Not(a bool) bool {
	return !a
}

func And(a, b bool) bool {
	return a == b
}

func Nand(a, b bool) bool {
	return Not(And(a, b))
}

func Nor(a, b bool) bool {
	return And(Not(a), Not(b))
}

func Or(a, b bool) bool {
	return Not(And(Not(a), Not(b)))
}

func Xor(a, b bool) bool {
	return Or(And(Not(a), b), And(a, Not(b)))
}

func Xnor(a, b bool) bool {
	return Not(Or(And(Not(a), b), And(a, Not(b))))
}

File: src/binary/gates/gates.js
function Not(a) {
	return !a
}

function And(a, b) {
	return a == b
}

function Nand(a, b) {
	return Not(And(a, b))
}

function Nor(a, b) {
	return And(Not(a), Not(b))
}

function Or(a, b) {
	return Not(And(Not(a), Not(b)))
}

function Xor(a, b) {
	return Or(And(Not(a), b), And(a, Not(b)))
}

function Xnor(a, b) {
	return Not(Or(And(Not(a), b), And(a, Not(b))))
}

const gates = {
    Not: Not,
    And: And,
    Nand: Nand,
    Nor: Nor,
    Or: Or,
    Xor: Xor,
    Xnor: Xnor
}

export { gates }